File: app/src/test/java/openfoodfacts/github/scrachx/openfood/utils/FileUtilsTest.java
package openfoodfacts.github.scrachx.openfood.utils;

import org.junit.Test;

import static com.google.common.truth.Truth.assertThat;

public class FileUtilsTest {
    private final String absoluteURL = "/path";
    private final String localURL = "file://path";

    @Test
    public void fileIsLocal_true() {
        assertThat(FileUtils.isLocaleFile(localURL)).isTrue();
    }

    @Test
    public void fileIsLocal_false() {
        assertThat(FileUtils.isLocaleFile(absoluteURL)).isFalse();
    }

    @Test
    public void isAbsolute_true() {
        assertThat(FileUtils.isAbsolute(absoluteURL)).isTrue();
    }

    @Test
    public void isAbsolute_false() {
        assertThat(FileUtils.isAbsolute(localURL)).isFalse();
    }
}

File: app/src/test/java/openfoodfacts/github/scrachx/openfood/utils/FileUtilsTest.kt
package openfoodfacts.github.scrachx.openfood.utils

import com.google.common.truth.Truth.assertThat
import org.junit.jupiter.api.Test

class FileUtilsTest {

    @Test
    fun `isLocaleFile returns true when file is local`() {
        assertThat(isLocaleFile(LOCAL_PATH)).isTrue()
    }

    @Test
    fun `isLocaleFile returns false when file is absolute`() {
        assertThat(isLocaleFile(ABSOLUTE_PATH)).isFalse()
    }

    @Test
    fun `isAbsoluteUrl returns true when file is local`() {
        assertThat(isAbsoluteUrl(LOCAL_PATH)).isFalse()
    }

    @Test
    fun `isAbsoluteUrl returns true when file is absolute`() {
        assertThat(isAbsoluteUrl(ABSOLUTE_PATH)).isTrue()
    }

    companion object {
        private const val ABSOLUTE_PATH = "/path"
        private const val LOCAL_PATH = "file://path"
    }
}